sql_request_abstract/models/sql_request_mixin.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (C) 2015 Akretion (<http://www.akretion.com>)
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import base64
+import logging
+import re
+import uuid
+from io import BytesIO
+
+from psycopg2 import ProgrammingError
+from psycopg2.sql import SQL
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SQLRequestMixin(models.AbstractModel):
+    _name = "sql.request.mixin"
+    _description = "SQL Request Mixin"
+
+    _clean_query_enabled = True
+
+    _check_prohibited_words_enabled = True
+
+    _check_execution_enabled = True
+
+    _sql_request_groups_relation = False
+
+    _sql_request_users_relation = False
+
+    STATE_SELECTION = [("draft", "Draft"), ("sql_valid", "SQL Valid")]
+
+    PROHIBITED_WORDS = [
+        "delete",
+        "drop",
+        "insert",
+        "alter",
+        "truncate",
+        "execute",
+        "create",
+        "update",
+        "ir_config_parameter",
+    ]
+
+    # Default Section
+    @api.model
+    def _default_group_ids(self):
+        ir_model_obj = self.env["ir.model.data"]
+        return [
+            ir_model_obj._xmlid_to_res_id("sql_request_abstract.group_sql_request_user")
+        ]
+
+    @api.model
+    def _default_user_ids(self):
+        return []
+
+    # Columns Section
+    name = fields.Char(required=True)
+
+    query = fields.Text(
+        required=True,
+        help="You can't use the following words"
+        ": DELETE, DROP, CREATE, INSERT, ALTER, TRUNCATE, EXECUTE, UPDATE.",
+    )
+
+    state = fields.Selection(
+        selection=STATE_SELECTION,
+        default="draft",
+        help="State of the Request:\n"
+        " * 'Draft': Not tested\n"
+        " * 'SQL Valid': SQL Request has been checked and is valid",
+    )
+
+    group_ids = fields.Many2many(
+        comodel_name="res.groups",
+        string="Allowed Groups",
+        relation=_sql_request_groups_relation,
+        column1="sql_id",
+        column2="group_id",
+        default=_default_group_ids,
+    )
+
+    user_ids = fields.Many2many(
+        comodel_name="res.users",
+        string="Allowed Users",
+        relation=_sql_request_users_relation,
+        column1="sql_id",
+        column2="user_id",
+        default=_default_user_ids,
+    )
+
+    has_group_changed = fields.Boolean(copy=False)
```

Comment:
Maybe some help here ?

Reply:
I would say, could you add an help string for this field? Thanks

Reply:
that's a technical field that is not displayed to the user. it mentions that the group changed since the last validation of the item.
i can add that text if you want ?

Reply:
Yes I think it could be nice because when you read code it became explained

Reply:
Done. Could you update your review ? 

thanks !